### PR TITLE
docs: build documentation using mkdocs

### LINKED ISSUE #233: Code documentation page
Let's move to [mkdocs](https://www.mkdocs.org/).

Make sure to update the github action accordingly

### PR DESCRIPTION
**Pull request by DaniBodor**
@JantineSmit, I created a draft pull request for your documentation building branch.
Once you feel it is ready (to be reviewed), you can click "Ready to review" below and tag me (and @psomhorst?) as a reviewer.

closes #233

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -0,0 +1,119 @@
+site_name: EITprocessing
+site_description: ALIVE project
+theme:
+  name: material
```
Comment (DaniBodor) [inline mkdocs.yml:4]: The `theme` key exists twice in this file, which shouldn't really happen in YML files. I am not sure which is read at the moment (this one is bare bones "material" theme, the one below has all kinds of additional details specified). Maybe test what it looks like when you remove either and stick to the one that you prefer.

File: pyproject.toml [hunk 2]
```diff
@@ -48,11 +48,14 @@ dev = [
...
     "coveralls",
-    "sphinx",
-    "sphinx_rtd_theme",
-    "sphinx-autoapi",
     "tox",
     "myst_parser",
+    "mkdocs",
+    "mkdocs-material",
+    "mkdocstrings",
+    "mkdocs-exlude",
```
Comment (DaniBodor) [inline pyproject.toml:56]: typo

```suggestion
    "mkdocs-exclude",
```

**Comment by psomhorst** [comment]
@JantineSmit What is the status of this PR? Is it ready for review? If so, can you click "Ready for review" (above them comment section at the bottom of the page) and mark me as reviewer?

**Comment by psomhorst** [comment]
I have taken the liberty to:

- rebase the documentation onto the newest version of main
- squash commits 
- add a [docs] dependency, removing the mkdocs and all dependencies from [dev]

**Comment by psomhorst** [comment]
I have updated the docs with:
- split the API into different categories
- streamlining README.md
- only show installation heading on installation page
- formatting of function signatures
- some layout improvements

**Review by psomhorst — APPROVED** [review]
I think this is a great effort that improves the documentation significantly. I have made some changes to make the API page more accessible. 

I have already deployed the new documentation to the github.io page. This should from now on be done automatically every time a release is made. There is no way to test this definitively without making a release, so we'll see whether that works the next time we make a release after this PR is merged.

@JantineSmit If you agree with these changes you can go ahead and merge the PR.

**Comment by psomhorst** [comment]
Great! Deployment of the documentation went as planned, so it should update every time we release a new version.

**Pull request merged**